doc/services/tfm/overview.rst

```diff
@@ -147,6 +147,11 @@ The default secure bootloader in TF-M is based on
 (for the second-stage bootloader, potentially after a HW-based bootloader on
 the secure MCU, etc.).
 
+TF-M also provides an optional immutable first stage bootloader - BL1. This can
+either be the default TF-M BL1 bootloader which uses most of the same codebase
+as BL2/MCUBoot mentioned above or a platform specific bootloader provided by a
+vendor.
```

Comment:
Are you certain it's supported for BL1 to point to a vendor specific bootloader?

How would I configure that?

Reply:
I'm concerned we are communicating that we support something that we don't.

Reply:
My understanding is that:
- TF-M provides a default bl1 implementation you can enable when building TF-M. This is in "bl1" directory in TF-M.
- Some of the platforms support a platform bl1 that they build as part of TF-M. This is for instance implemented in  "platform/ext/target/arm/corstone1000."

Reply:
@SebastianBoe Any further thoughts here?

I'm hesistant to make it easy for people to change BL1, if the real goal is to make BL2 updateable, when my understanding of TF-M is that BL2 is the actual root of trust here and must be immutable. The documentation changes in this PR point to verifying the signature or BL2, but that's just pushing the root of trust one level lower. Is the real goal making BL2 updateable? That's not an invalid approach ... but I think it conflicts with the intent and promises TF-M is making about where the root of trust it.